### PR TITLE
feat(enhancement): Hide less important scan messages

### PR DESCRIPTION
**Pull request by TomGoodIdea**
**Feature:** This PR implements the feature request detailed and discussed in issue #7932; I chose https://github.com/endless-sky/endless-sky/issues/7932#issuecomment-1364656501 as the best solution.

## Feature Details
Hides all messages about scans performed on player's ships, except when the target is either
 - the player's flagship
 - carrying cargo or outfits considered illegal by the government performing the scan

## Testing Done
Went to a Republic system and got flagship and escorts scanned, some having illegal outfits installed.

## Performance Impact
None.

### CODE REVIEW

**Review by mOctave — APPROVED** [review]
Code looks fine, although I haven't actually tested it.

**Comment by warp-core** [comment]
Is there any reason to ever actually show the "unimportant" scan messages at all?

File: source/Ship.cpp [hunk 1]
```diff
@@ -2710,6 +2710,22 @@ int Ship::Scan(const PlayerInfo &player)
 	if(isYours || (target->isYours && activeScanning))
 		Audio::Play(Audio::Get("scan"), Position());
 
+	bool isImportant;
```
Comment (warp-core) [inline source/Ship.cpp:2713]: Always initialise primitives.
```suggestion
	bool isImportant = false;
```

**Comment by TomGoodIdea** [comment]
Idk what was the reason for adding them, but they're pretty old (40d87c4) and some people may find them useful.
I'm not sure about removing them completely, but we can make the new setting default to on, or leave it off and change the name to "Show extra scan messages".

**Comment by Amazinite** [comment]
As warp points out, there not really any reason to see scans that don't matter. I'd say just make this the default behavior, not having a preference to toggle it.

> some people may find them useful.

Who and why? "Some may" is a bit of a weasel phrase. If we want to keep the ability to see these messages, then I think we need a good argument as to why, not just shrugging and saying "It might be useful to someone somewhere."

**Comment by mOctave** [comment]
I think we should keep, but have this preference default to on. They could be useful for debugging and development, after all.

**Comment by quyykk** [comment]
> They could be useful for debugging and development, after all.

I don't think this is a good reason. For development it is always possible to readd the messages if they are needed, or better yet use a debugger :smile:

I also agree that this shouldn't be a preference.

File: source/Ship.cpp [hunk 2]
```diff
@@ -2710,6 +2710,21 @@ int Ship::Scan(const PlayerInfo &player)
...
+			for(auto &it : target->Outfits())
+				if(government->Fines(it.first) || government->Condemns(it.first))
+				{
+					hasIllegal = true;
+					break;
+				}
+
+		isImportant = hasIllegal || target.get() == player.Flagship();
+	}
+
```
Comment (warp-core) [inline source/Ship.cpp:2727]: I feel like this stuff should be done somewhere else. In Government or maybe a new class.
Adding this here is another place that needs to be updated if the way fining works is changed, like what #8677 tries to do.

Comment (TomGoodIdea) [reply source/Ship.cpp]: done

File: source/Ship.cpp [hunk 3]
```diff
@@ -2719,6 +2719,10 @@ int Ship::Scan(const PlayerInfo &player)
 	if(isYours || (target->isYours && activeScanning))
 		Audio::Play(Audio::Get("scan"), Position());
 
+	bool isImportant = false;
+	if(target->isYours)
+		isImportant = government->FinesContents(target.get()) || target.get() == player.Flagship();
```
Comment (warp-core) [inline source/Ship.cpp:2724]: Left operand is evaluated first and the pointer comparison is going to be way faster than the other thing.
```suggestion
		isImportant = target.get() == player.Flagship() || government->FinesContents(target.get());
```

**Pull request merged**